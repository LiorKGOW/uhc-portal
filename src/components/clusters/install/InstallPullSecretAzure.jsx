import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  PageSection, Card, Title, Alert,
} from '@patternfly/react-core';
import PageTitle from '../../common/PageTitle';

import TokenErrorAlert from './instructions/components/TokenErrorAlert';
import PullSecretSection from './instructions/components/PullSecretSection';

import { tollboothActions } from '../../../redux/actions';
import { scrollToTop } from '../../../common/helpers';
import ExternalLink from '../../common/ExternalLink';

class InstallPullSecretAzure extends Component {
  componentDidMount() {
    scrollToTop();
    document.title = 'Install OpenShift 4 | Pull Secret';

    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const azureText = 'Download or copy your pull secret. Your pull secret provides your Azure Red Hat OpenShift cluster with access to additional content provided by Red Hat.';
    const msDocLink = 'https://docs.microsoft.com/en-us/azure/openshift/howto-add-update-pull-secret';
    return (
      <>
        <PageTitle title="Azure Red Hat OpenShift" />
        <PageSection className="ocp-instructions">
          <>
            <Title headingLevel="h3" size="2xl">
              Red Hat content access
            </Title>
            <Card>
              <div className="pf-l-grid pf-m-gutter ocm-page">
                {token.error && <TokenErrorAlert token={token} />}
                <div className="pf-c-content">
                  <h3 className="pf-c-title pf-m-md downloads-subtitle">Pull secret</h3>
                  <PullSecretSection token={token} text={azureText} />
                  <Alert
                    variant="info"
                    isInline
                    title="Connecting your cluster to OpenShift Cluster Manager"
                    className="bottom-alert"
                  >
                    <p>
                      Azure Red Hat OpenShift clusters do not connect to OpenShift Cluster Manager
                      at creation time. Follow the
                      {' '}
                      <ExternalLink href={msDocLink}>
                        Microsoft documentation
                      </ExternalLink>
                      {' '}
                      if you would like to enable telemetry as a day 2 operation.
                    </p>
                  </Alert>
                </div>
              </div>
            </Card>
          </>
        </PageSection>
      </>
    );
  }
}

InstallPullSecretAzure.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ token: state.tollbooth.token });

export { InstallPullSecretAzure };

export default connect(mapStateToProps)(InstallPullSecretAzure);
