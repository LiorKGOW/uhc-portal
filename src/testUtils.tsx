import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
