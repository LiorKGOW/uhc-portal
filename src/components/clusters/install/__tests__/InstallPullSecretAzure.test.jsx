import React from 'react';
import { screen, checkAccessibility, TestRouter, withState } from '~/testUtils';

import { InstallPullSecretAzure } from '../InstallPullSecretAzure';
import githubReleases from '../githubReleases.mock';

jest.mock('../../../../redux/actions', () => ({
  __esModule: true,
  tollboothActions: {
    createAuthToken: jest.fn().mockResolvedValue('foo'),
  },
  githubActions: {
    getLatestRelease: jest.fn(),
  },
}));

describe('<InstallPullSecretAzure />', () => {
  const dispatch = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it.skip('is accessible', async () => {
    const { container } = withState(githubReleases).render(
      <TestRouter>
        <InstallPullSecretAzure token={{}} dispatch={dispatch} />
      </TestRouter>,
    );

    expect(await screen.findByText('Azure Red Hat OpenShift')).toBeInTheDocument();

    // This fails due to a "Heading levels should only increase by one (heading-order)" error
    await checkAccessibility(container);
  });
});
