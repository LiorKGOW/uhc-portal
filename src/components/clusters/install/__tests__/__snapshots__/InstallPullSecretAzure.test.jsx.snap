// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallPullSecretAzure renders correctly 1`] = `
<AppPage
  title="Install OpenShift 4 | Pull Secret"
>
  <PageTitle
    title="Azure Red Hat OpenShift"
  />
  <PageSection
    className="ocp-instructions"
  >
    <Title
      headingLevel="h3"
      size="2xl"
    >
      Red Hat content access
    </Title>
    <Card>
      <div
        className="pf-l-grid pf-m-gutter ocm-page"
      >
        <div
          className="pf-c-content"
        >
          <h3
            className="pf-c-title pf-m-md downloads-subtitle"
          >
            Pull secret
          </h3>
          <PullSecretSection
            text="Download or copy your pull secret. Your pull secret provides your Azure Red Hat OpenShift cluster with access to additional content provided by Red Hat."
            token={Object {}}
          />
          <Alert
            className="bottom-alert"
            isInline={true}
            title="Connecting your cluster to OpenShift Cluster Manager"
            variant="info"
          >
            <p>
              Azure Red Hat OpenShift clusters do not connect to OpenShift Cluster Manager at creation time. Follow the
               
              <ExternalLink
                href="https://docs.microsoft.com/en-us/azure/openshift/howto-add-update-pull-secret"
              >
                Microsoft documentation
              </ExternalLink>
               if you would like to enable telemetry as a day 2 operation.
            </p>
          </Alert>
        </div>
      </div>
    </Card>
  </PageSection>
</AppPage>
`;
